src/jit/rangecheck.cpp

```diff
@@ -825,30 +866,31 @@ Range RangeCheck::ComputeRangeForLocalDef(BasicBlock*          block,
                                           GenTreeLclVarCommon* lcl,
                                           bool monotonic DEBUGARG(int indent))
 {
-    // Get the program location of the def.
-    Location* loc = GetDef(lcl);
-
-    // If we can't reach the def, then return unknown range.
-    if (loc == nullptr)
+    BasicBlock* asgBlock;
+    GenTreeOp*  asg = GetSsaDefAsg(lcl, &asgBlock);
+    if (asg == nullptr)
     {
         return Range(Limit(Limit::keUnknown));
     }
 #ifdef DEBUG
+    // Get the program location of the def.
+    Location* loc = GetDef(lcl);
```

Comment:
Looks like you're repeating these assertions after every call to `GetSsaDefAsg`; maybe move them down into `GetSsaDefAsg` itself?

Reply:
Good idea, I'll try it. The only potential problem is that JITDUMP that needs the location, maybe I'll just throw it away.